src/main/sensors/opticalflow.c

```diff
@@ -0,0 +1,200 @@
+            }
+#endif
+            break;
+
+        case OPTICALFLOW_NONE:
+            opticalflowHardware = OPTICALFLOW_NONE;
+            break;
+    }
+
+    if (opticalflowHardware == OPTICALFLOW_NONE) {
+        sensorsClear(SENSOR_OPTICALFLOW);
+        return false;
+    }
+
+    detectedSensors[SENSOR_INDEX_OPTICALFLOW] = opticalflowHardware;
+    sensorsSet(SENSOR_OPTICALFLOW);
+    return true;    
+}
+
+bool opticalflowInit(void) {
+    if (!opticalflowDetect(&opticalflow.dev, opticalflowConfig()->opticalflow_hardware)) {
+        return false;
+    }
+
+    opticalflow.dev.init(&opticalflow.dev);
+    opticalflow.quality = OPTICALFLOW_NO_NEW_DATA;
+    opticalflow.rawFlowRates.X = 0;
+    opticalflow.rawFlowRates.Y = 0;
+    opticalflow.processedFlowRates.X = 0;
+    opticalflow.processedFlowRates.Y = 0;
+    opticalflow.lastValidResponseTimeMs = millis();
+
+    if (opticalflowConfig()->rotation != 0) {
+        cosRotAngle = cosf(DEGREES_TO_RADIANS(opticalflowConfig()->rotation));
+        sinRotAngle = sinf(DEGREES_TO_RADIANS(opticalflowConfig()->rotation));
+    }
+    //low pass filter
+    if (opticalflowConfig()->flow_lpf != 0) {
+        const float flowCutoffHz = (float)opticalflowConfig()->flow_lpf / 100.0f;
+        const float flowGain     = pt2FilterGain(flowCutoffHz, opticalflow.dev.delayMs / 1000.0f);
+
+        pt2FilterInit(&xFlowLpf, flowGain);
+        pt2FilterInit(&yFlowLpf, flowGain);
+    }
+    return true;
+}
+
+void opticalflowUpdate(void) {
+    if (opticalflow.dev.update) {
+        opticalflow.dev.update(&opticalflow.dev);
+    }
+}
+
+void opticalflowProcess(void) {
+    opticalflowData_t data = {0};
+    bool hasNewData = opticalflow.dev.read(&opticalflow.dev, &data);
+    
+    opticalflow.quality = data.quality;
+    data.devMinQualityThreshold = opticalflow.dev.minQualityThreshold;
+
+    if (hasNewData) {
+        opticalflow.lastValidResponseTimeMs = millis();
+
+        opticalflow.rawFlowRates = data.flowRate;
+        opticalflow.deltaTimeUs  = data.deltaTimeUs;
+        
+        applySensorRotation(&opticalflow.processedFlowRates, &opticalflow.rawFlowRates);
+
+        applyLPF(&opticalflow.processedFlowRates);
+
+        // DEBUG SECTION
+        DEBUG_SET(DEBUG_OPTICALFLOW, 0, opticalflow.quality);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 1, opticalflow.rawFlowRates.X);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 2, opticalflow.rawFlowRates.Y);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 3, opticalflow.processedFlowRates.X);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 4, opticalflow.processedFlowRates.Y);
+        DEBUG_SET(DEBUG_OPTICALFLOW, 5, (int32_t)opticalflow.deltaTimeUs);
+    }
+}
+
+static void applySensorRotation(opticalflowRates_t * dist, opticalflowRates_t * src) {
+    dist->X = (int32_t)(src->X * cosRotAngle - src->Y * sinRotAngle);
+    dist->Y = (int32_t)(src->X * sinRotAngle + src->Y * cosRotAngle);
+    
+    if (opticalflowConfig()->flipY) {
+        dist->Y = -dist->Y;
```

Comment:
Personally, I'd use 'flip first, around X (y=-y), then rotate`. But this should match sensor orientation of other sensors, so it is rotate first, than flip around Y (x=-x)
![bs_rotations](https://github.com/user-attachments/assets/974c14b0-0e8c-406e-a5fe-d16c7c040f64)